src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -632,17 +632,30 @@ namespace
                 "medium_priority",
                 get_medium_priority(instance_node->GetObjectRef(), time));
         if (type == RenderType::MaterialPreview)
+        {
             params.insert_path("visibility.shadow", false);
-
-        // Create the instance and insert it into the assembly.
-        assembly.object_instances().insert(
-            asr::ObjectInstanceFactory::create(
-                instance_name.c_str(),
-                params,
-                object_info.m_name.c_str(),
-                transform,
-                front_material_mappings,
-                back_material_mappings));
+            // Create the instance and insert it into the assembly.
```

Comment:
This is a workaround. We must understand what caused the problem in the first place, maybe there's a bug somewhere else and I would rather fix the original problem than hack around it.

That said, the rest of the PR looks good so I would suggest to remove this part and keep (and merge) the rest.

Thanks!

Reply:
Revert indentation.